lib/grammers-session/src/message_box/mod.rs

```diff
@@ -278,10 +277,22 @@ impl MessageBox {
         });
     }
 
-    /// Begin getting difference for the given entry.
+    /// Try to begin getting difference for the given entry.
+    /// Fails if the entry does not have a previously-known state that can be used to get its difference.
     ///
     /// Clears any previous gaps.
-    fn begin_get_diff(&mut self, entry: Entry) {
+    fn try_begin_get_diff(&mut self, entry: Entry) {
+        if self.map.get(&entry).is_none() {
```

Comment:
I'm guessing this and the check below can both use [`contains_key`](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html) (but it's fine to leave it as-is; both should work the same).

Reply:
Done